feat: Snowflake Open Catalog for Iceberg sink connector (#841)
aiven/aiven-docs: docs/products/kafka/howto/snowflake-open-catalog.md

Use the Iceberg sink connector in Aiven for Apache Kafka® Connect to write data to
Iceberg tables managed by Snowflake Open Catalog.

## Prerequisites

- A [Snowflake Open Catalog account](https://other-docs.snowflake.com/en/opencatalog/create-open-catalog-account)
- A Polaris catalog created using
  [Amazon S3 as the storage backend](https://other-docs.snowflake.com/en/opencatalog/create-catalog#create-a-catalog-using-amazon-simple-storage-service-amazon-s3)
- An [Aiven for Apache Kafka® service](/docs/products/kafka/kafka-connect/howto/enable-connect)
  with Apache Kafka Connect enabled, or
  a [dedicated Aiven for Apache Kafka Connect® service](/docs/products/kafka/kafka-connect/get-started#apache_kafka_connect_dedicated_cluster)
- An Apache Kafka topic that contains data to sink
- An Apache Kafka control topic (default: `control-iceberg`)
- AWS S3 credentials and a configured S3 bucket
- Polaris authentication credentials
  - `iceberg.catalog.scope`: A role scope defined in the Open Catalog
  - `iceberg.catalog.credential`: A token or hash for authenticating with the catalog

:::note

> Snowflake Open Catalog does not support automatic catalog creation. You must create the
> catalog manually in the Snowflake console.

Reviewer: Consider a bit shorter version and adding a link, e.g.:

```suggestion
Snowflake Open Catalog does not support automatic catalog creation. [Create the
catalog](https://other-docs.snowflake.com/en/opencatalog/create-catalog) manually in the Snowflake console.
```

:::

## Create an Iceberg sink connector configuration

To configure the Iceberg sink connector, define a JSON configuration with the required catalog, S3, and Kafka settings.

:::note
Loading worker properties is not supported. Use `iceberg.kafka.*` properties instead.
:::

```json
{

>   "name": "<your-connector-name>",

Reviewer: This comment refers to all placeholder instances on this PR:
Normally, we use ALL-CAPS for placeholders in the Aiven docs. Please considers using the following for consistency:

```suggestion
  "name": "CONNECTOR_NAME",
```
and relevant alternatives for all the other placeholders.

- `iceberg.control.commit.timeout-ms`: Max time (in ms) to wait for a commit
- `iceberg.control.topic`: Control topic used by Iceberg (default: `control-iceberg`)
- `iceberg.catalog.type`: Set to `rest` for Snowflake Open Catalog
- `iceberg.catalog.uri`: Polaris REST catalog endpoint (from the Snowflake Open Catalog
  console)
- `iceberg.catalog.scope`: Role scope in the format `PRINCIPAL_ROLE:<role_name>`
- `iceberg.catalog.credential`: Authentication credential for Snowflake Open Catalog
  Use the format `<client_id>:<secret>` from the configured service connection that
  uses a Principal role
- `iceberg.catalog.warehouse`: Name of the catalog created in Snowflake Open Catalog.
  This is not the S3 bucket name
- `iceberg.catalog.client.region`: AWS region of the S3 bucket
- `iceberg.catalog.io-impl`: Set to `org.apache.iceberg.aws.s3.S3FileIO`
- `iceberg.catalog.s3.access-key-id`: AWS access key ID with write permissions
- `iceberg.catalog.s3.secret-access-key`: AWS secret access key
- `iceberg.catalog.s3.path-style-access`: Enable (`true`) or disable (`false`)
  path-style access

## Create the Iceberg sink connector

<CreateIcebergSinkConnector />

> ## Example: Define and create an Iceberg sink connector

Reviewer: Consider a shorter variant:

```suggestion
## Example
```

> This example shows a full configuration using Polaris:

Reviewer: ```suggestion
This example shows a full configuration using Polaris to create an Iceberg sink connector: 
```

  "iceberg.catalog.io-impl": "org.apache.iceberg.aws.s3.S3FileIO",
  "iceberg.catalog.s3.access-key-id": "your-access-key-id",
  "iceberg.catalog.s3.secret-access-key": "your-secret-access-key",
  "iceberg.catalog.warehouse": "my-bucket",
  "iceberg.tables": "mydatabase.mytable",
  "iceberg.tables.auto-create-enabled": "true",
  "iceberg.control.commit.interval-ms": "1000",
  "iceberg.control.commit.timeout-ms": "20000",
  "key.converter": "org.apache.kafka.connect.json.JsonConverter",
  "value.converter": "org.apache.kafka.connect.json.JsonConverter",
  "iceberg.kafka.bootstrap.servers": "kafka.example.com:9092",
  "iceberg.kafka.security.protocol": "SSL",
  "iceberg.kafka.ssl.keystore.location": "/run/aiven/keys/public.keystore.p12",
  "iceberg.kafka.ssl.keystore.password": "password",
  "iceberg.kafka.ssl.keystore.type": "PKCS12",
  "iceberg.kafka.ssl.truststore.location": "/run/aiven/keys/public.truststore.jks",
  "iceberg.kafka.ssl.truststore.password": "password",
  "iceberg.kafka.ssl.key.password": "password"
}
```

<RelatedPages/>

- [Iceberg sink connector overview](/docs/products/kafka/kafka-connect/howto/iceberg-sink-connector)

> - [AWS Glue REST catalog](/docs/products/kafka/kafka-connect/howto/aws-glue-rest-catalog)

Reviewer 2: Optionally remove this line!
```suggestion
```
